### ISSUE TITLE
Setting vsync on Intel integrated has validation error

### ISSUE THREAD

**Issue by m4b** [gfx-rs/wgpu#350]
when i switch to NoVsync i get this validation warning (it does appear however to not vsync anymore):

```
VALIDATION [VUID-VkSwapchainCreateInfoKHR-presentMode-01281 (0)] : vkCreateSwapchainKHR() called with a non-supported presentMode (i.e. VK_PRESENT_MODE_IMMEDIATE_KHR). The Vulkan spec states: presentMode must be one of the VkPresentModeKHR values returned by vkGetPhysicalDeviceSurfacePresentModesKHR for the surface (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-VkSwapchainCreateInfoKHR-presentMode-01281)
object info: (type: DEVICE, hndl: 94235621371968)
```

This is on archlinux with intel integrated card

**Comment by m4b** [comment]
And this is on wgpu 0.3.3 crates io version

**Comment by kvark** [comment]
Basically, we need to consider the available presentation modes when choosing one. In the worst case, we keep Vsync ON.

**Comment by m4b** [comment]
So what is strange to me is how could it not be vsyncing now if my card is reporting it cannot be put in a NON vsync mode ?

**Comment by kvark** [comment]
Found the relevant TODO: https://github.com/gfx-rs/wgpu/blob/78fbbba5e928e4a6f15a03a1c04d06916a947673/wgpu-native/src/swap_chain.rs#L98

**Comment by kvark** [comment]
The supported present are obtained here: https://github.com/gfx-rs/wgpu/blob/78fbbba5e928e4a6f15a03a1c04d06916a947673/wgpu-native/src/device.rs#L1987

You can add a `info!` log to print these out and see.

**Comment by paulkernfeld** [comment]
I think this was fixed by #424.
